lib/topology/platforms/shell.py

```diff
@@ -362,15 +362,17 @@ def send_command(
         See :meth:`BaseShell.send_command` for more information.
         """
         # Get connection
+        connected = False
         try:
-            spawn = self._get_connection(connection)
-
+            connected = self.is_connected(connection)
         except NonExistingConnectionError:
             if not self._auto_connect:
                 raise
 
+        if not connected:
```

Comment:
This is what I mean when I doubt the purpose of ``_auto_connect``. If ``_auto_connect`` is set to ``False`` an exception will be raised when the user tries to call ``send_command`` while using a ``connection`` attribute set to a connection that does not exist yet. This can warn the user that he or she is trying to perform a ``send_command`` to a connection that does not exist yet, but, the user will already know that since he or she has never called ``connect`` with that ``connection`` value before.

That is why I see little value in this attribute, if that exception is raised, the user will just handle it to call ``connect`` with the same ``connection`` attribute which at the end is just the same thing as not having this ``auto_connect`` attribute at all and always attempt to call ``connect`` when a non-existing ``connection`` attribute is used in ``send_command``.

Also, if you want to use ``_auto_connect`` you should perform this same check here.